packages/expo-screen-capture/src/ScreenCapture.ts

```diff
@@ -0,0 +1,52 @@
+import { useEffect } from 'react';
+
+import ExpoScreenCapture from './ExpoScreenCapture';
+
+/**
+ * Prevents screenshots and screen recordings. If you are
+ * already preventing screen capture, this method does nothing.
+ *
+ * On iOS, this will only prevent screen recordings, and is only
+ * available on iOS 11 and newer. On older iOS versions, this method
+ * does nothing.
+ *
+ * @example
+ * ```typescript
+ * preventScreenCapture();
+ * ```
+ */
+export async function preventScreenCapture() {
+  return ExpoScreenCapture.preventScreenCapture();
+}
+
+/**
+ * Reallows screenshots and recordings. If you haven't called
+ * `preventScreenCapture()` yet, this method does nothing.
+ *
+ * @example
+ * ```typescript
+ * allowScreenCapture();
+ * ```
+ */
+export async function allowScreenCapture() {
+  return ExpoScreenCapture.allowScreenCapture();
+}
+
+/**
+ * React hook for preventing screenshots and screen recordings
+ * while the component is mounted.
+ *
+ * @example
+ * ```typescript
+ * usePreventScreenCapture();
+ * ```
+ */
+export function usePreventScreenCapture(): void {
+  useEffect(() => {
+    preventScreenCapture();
+
+    return () => {
+      allowScreenCapture();
+    };
+  }, []);
+}
```

Comment:
I noticed one bug with this, which would be something good to test. Namely, if two simultaneously components use this hook and one is unmounted, suddenly screen capture will be allowed even though the other component is still mounted. Can you think of a clean way to address this? What about when someone mounts a component with this hook and separately calls the imperative methods? What behavior is intuitive for developers and simple for us to implement? (Once we decide, we also should document this. This is the stuff that's actually important in API documentation. Things like parameters and return values are just the surface.)

Reply:
We can add a 'label' parameter to the `prevent` and `allow` calls (and, by extension, the hook). If you are going to have multiple components up using the hook, or combining cases where you're using both the imperative methods _and_ the hook, you'd add labels to your calls to make sure they don't overwrite each other.

```
// add your component name (or some other unique identifier) to the hook call so that other
// components using the hook don't remove clash
usePreventScreenCapture("SomeComponentName")

...

preventScreenCapture("lol");
preventScreenCapture("omg");

allowScreenCapture("lol");
// screen capture is still blocked
```

`label` defaults to "default"

Reply:
The most recent commits address this

We now maintain a set of the above-described `tags` and keep track of the user adding/removing them as they call `preventScreenCaptureAsync` and `allowScreenCaptureAsync`. If they call `allowScreenCaptureAsync` and (after removing that tag) there are zero entries in the set, we call the native methods to re-allow screen capturing.